cirq-core/cirq/optimizers/merge_interactions_to_sqrt_iswap_test.py

```diff
@@ -0,0 +1,187 @@
+from typing import Callable, List
+
+import cirq
+
+
+def assert_optimizes(before: cirq.Circuit, expected: cirq.Circuit, **kwargs):
+    actual = cirq.Circuit(before)
+    opt = cirq.MergeInteractionsToSqrtIswap(**kwargs)
+    opt.optimize_circuit(actual)
+
+    # Ignore differences that would be caught by follow-up optimizations.
+    followup_optimizations: List[Callable[[cirq.Circuit], None]] = [
+        cirq.merge_single_qubit_gates_into_phased_x_z,
+        cirq.EjectPhasedPaulis().optimize_circuit,
+        cirq.EjectZ().optimize_circuit,
+        cirq.DropNegligible().optimize_circuit,
+        cirq.DropEmptyMoments().optimize_circuit,
+    ]
```

Comment:
What's the idea behind testing the behavior caused by other follow-up optimizations here? I think this has the risk of making the unit tests brittle. Maybe we should have a separate place for such integration tests where we actually prescribe this sequence of optimizations and then test their combined behavior?

Reply:
I copied most of the tests from `merge_interactions_test.py`.  I think the test is trying to be *less* brittle to changes in the two-qubit synthesis method by relying on more basic optimizers.